docs/DeveloperGuide.md

```diff
@@ -494,6 +494,52 @@ The below UML diagram represents the overall structure of the predicates for `Bu
 4. Filtering entries by name - that is, providing the `-n` flag to the filter command, will filter all entries whose names
    contain the parameter provided to `-n` as a *substring*.
 
+### Sorting buyers and properties
+
+To sort `Buyers` and `Properties`, the `ObservableArrayList` that stores references to these objects and displays them on the user's screen
+is modified directly to a sorted version. These changes are propagated directly to the `FilteredList`, enabling users to sort
+a previously filtered list. As the `FilteredList` is based on the `ObservableArrayList`, users can also sort the list first,
+then filter it. This results in users being able to build sort and filter functions on top of each other to more powerfully
+manipulate the list based on their needs.
+
+A `Comparator` is used to sort the `ObservableArrayList`. Different comparators with different conditions are used to sort the
+list by different criteria. The following are the `Comparators` used to allow for the corresponding sorting functions:
+`Buyer`: `BuyerComparator`
+1. `NameComparator`
+2. `PriceRange Comparator`
+3. `Priority Comparator`
+`Property`: `PropertyComparator`
+4. `PropertyNameComparator`
+5. `PriceComparator`
+Both:
+6. `TimeComparator`
+
+The UML diagrams below represents the overall structure of the `Comparator`s used.
+
+![BuyerComparatorsClassDiagram](images/SortBuyerComparatorsClassDiagram.png)
+![PropertyComparatorsClassDiagram](images/SortPropComparatorsClassDiagram.png)
+
+#### Design Considerations
+Similar to the `FilteredList` abstraction provided by JavaFX, we considered using a `SortedList` to present the list in a
+sorted version without modifying the underlying data structure `ObservableArrayList`. This is
+to preserve the chronological order in which users enter the entries so that it can still be displayed with the `list` command.
+
+However, this meant that we needed to have both `FilteredList` and `SortedList` stored and vary which one is displayed to users
+on entering a command. As such, the last-shown list changes depending on the last command entered. To keep track of this,
+we used a flag which would be updated everytime a `filter` or `sort` command was used. All `Command`s that referred to an
+entry on the displayed list were adapted to take relative indices from the last-shown list indicated by the flag. The `Model`
+component also needed access to the `UI` component's `BuyerPanelList` and `PropertyPanelList` in order to display
+the corresponding `FilteredList` or `SortedList` based on changes to the flag. To reduce coupling, we would have had to
+apply the Observer pattern.
+
+We decided against this as the complicated design made it more bug-prone. In addition, it did not allow for
+stacking of `filter` and `sort` functions, that is, a user is unable to filter on top of a sorted list or vice versa as the
+`FilteredList` and `SortedList` are independent and separate from each other.
```

Comment:
Perhaps this portion of the writeup could be included under design considerations, following the structure of other parts of the dg above!

Reply:
What do you mean haha it's under the design considerations section! Or did you mean something else?